admin-api/admin-api-specification-v1.0.md

```diff
@@ -1,4 +1,906 @@
+
+The `GET /transactions/{id}` request sends an empty request body.
+
+The following data is returned in the case of a successful operation:
+
+| Name  | Required  | Type | Description |
+|---|---|--|--|
+| `quoteId`  |   | [UUID](#uuid) | Quote identifier. |
+| `transactionId`  |   | [UUID](#uuid) | Transaction identifier. |
+| `transactionRequestId`  |   | [String](#string) | Identifies an optional previously-sent transaction request. |
+| `payee`  |   | [Party](#party) | Payee details. |
+| `payer`  |   | [Party](#party) | Payer details. |
+| `amount`  |   | [Money](#money) | Transaction amount. |
+| `transactionType`  |   | [TransactionType](#transactiontype) | Transaction details. |
+| `note`  |   | [String](#string) | A memo that will be attached to the transaction. |
+| `extensionList`  |   | [ExtensionList](#extensionlist) | Additional details. |
+
+# Data models used by the API
+
+## Format
+
+For details on the formats used for element data types used by the API, see section [7.1 Format Introduction](../fspiop-api/documents/API-Definition_v1.1.md#71-format-introduction) in the Mojaloop FSPIOP API Definition.
+
+## Element data type formats
+
+This section defines element data types used by the API.
+
+### Amount
+
+For details, see section [7.2.13 Amount](../fspiop-api/documents/API-Definition_v1.1.md#7213-amount) in the Mojaloop FSPIOP API Definition.
+
+### Boolean
+
+A `"true"` or `"false"` value.
+
+### DateTime
+
+For details, see section [7.2.14 DateTime](../fspiop-api/documents/API-Definition_v1.1.md#7214-datetime) in the Mojaloop FSPIOP API Definition.
+
+### Enum
+
+For details, see section [7.2.2 Enum](../fspiop-api/documents/API-Definition_v1.1.md#722-enum) in the Mojaloop FSPIOP API Definition.
+
+### Integer
+
+For details, see section [7.2.5 Integer](../fspiop-api/documents/API-Definition_v1.1.md#725-integer) in the Mojaloop FSPIOP API Definition.
+
+### Number
+
+The API data type **Number** is a an arbitrary-precision, base-10 decimal number value.
+
+### String
+
+For details, see section [7.2.1 String](../fspiop-api/documents/API-Definition_v1.1.md#721-string) in the Mojaloop FSPIOP API Definition.
+
+### UUID
+
+For details, see section [7.2.16 UUID](../fspiop-api/documents/API-Definition_v1.1.md#7216-uuid) in the Mojaloop FSPIOP API Definition.
+
+## Element definitions
+
+This section defines element types used by the API.
+
+### AutoPositionReset
+
+Data model for the element **AutoPositionReset**.
+
+| Name  | Required  | Type | Description |
+|---|---|--|--|
+| `autoPositionReset`  | yes | [Boolean](#boolean) | A flag to indicate whether or not a settlement model requires the automatic reset of the position. |
+
+### IsActive
+
+Data model for the element **IsActive**.
+
+| Name  | Required  | Type | Description |
+|---|---|--|--|
+| `isActive`  | yes | [Integer(1)](#integer) | A flag to indicate whether or not a ledger account / participant is active. Possible values are `1` (active) and `0` (not active). |

```

Comment:
Why do we have `IsActive` and `IsActiveBoolean`? That seems pretty nasty to me.

Reply:
The API sometimes takes the active status to be indicated by `1`, and at other times by `true`. A distinction between the two had to be made. The body of the `PUT /participants/{name}` sends `"isActive": true`, whereas the response gives you `"isActive": 1`.

Reply:
Yeah this isn't great API design. If we have a term such as `isActive`, it's data type should be consistent. Why can't we make them consistent?